### DISCUSSION TITLE
Understanding the "delay or packet loss is too high!" warning

### DISCUSSION THREAD

**Discussion by Aidan63** [OvenMediaLabs/OvenMediaEngine#1545] [Help] [answered]
Hello, another question (sorry).

I've been trying out the webrtc tester and have been running into the "[warning] delay or packet loss is too high!" message with just one input and one client. I've tried with the tester on a separate machine from ome and the same machine as well trying with my local ip and public one, in all cases I constantly see these warnings.

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/10246746/bb1a4746-ee09-4bbf-b3b0-b9dacdf1a4ad)

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/10246746/e0608b70-e0f1-4225-bcf0-578aacd40587)

The docs seem to imply this usually happens when the server machine is overloaded, but looking at htop shows no ome related process above 10% usage and all threads are sat at around 20% usuage, so no problem. This machine is running a Ryzen 7 2700x so I wouldn't expect it to have any problems with just a single stream in and a single client connected (albeit creating three varying stream qualities).

To avoid any weird container resource limits this was outside a container. i.e. I compiled ome from source and ran it manually.

Unless I'm misunderstanding the video delay metric I don't actually have the 1s delay it mentions. In my setup the web browser can send commands over a web socket to my custom provisioner client to pause the video, next frame, prev frame, playback speed, etc, and these all seem to effect the received video stream almost immediately (I have not measured it but its definitely not near 1s). While the stream doesn't appear to be delayed it is choppier than I'd expect and the 15fps that report mentions sounds like it could be right.

I tried tweaking some of the worker counts mentioned in the docs but I haven't noticed any real change. Am I misinterpreting this message and if so what does it mean in my case where the server doesn't appear to be over loaded?

[Server.xml](https://gist.github.com/Aidan63/53f66d9bc3c7826c9417b705d4f0bcdf)

Thanks for all the help.

**Comment by basisbit** [comment]
What tool do you use to create those log outputs? What is used for streaming, what is used for playback?

**Comment by getroot** [comment, marked as answer]
The principle of measuring dealy in OvenRtcTester is to measure it by comparing the wall clock time and the increase in RTMP timestamp. If there are sufficient resources on the OME side, it would be a good idea to check whether there is a delay on the input stream side or whether the RTP timestamp is normal.

**Reply by Aidan63** [reply]
Thanks, issue seems to be that I need to manually provide packet durations in RTP units in my client and I was doing a bad job of it. Made some progress on that and it seems to have improved things.
